### ISSUE TITLE
Error: COM is not supported

### ISSUE THREAD

**Issue by Fede-26** [vignette-project/osu-framework-camera#7]
I tried to build the source with `dotnet build -v diag --no-cache /p:CubismFrameworkRuntime=linux-x64 /p:GenerateFullPaths=true /t:build /consoleloggerparameters:NoSummary` and it compiled succesfully.

When I try to run with `dotnet run --project holotrack.Desktop` or `dotnet holotrack.Desktop/bin/Debug/netcoreapp3.1/holotrack.Desktop.dll ` it gives me:

```
Unhandled exception. System.PlatformNotSupportedException: COM is not supported
at DirectShowLib.DsDevice.GetDevicesOfCat(Guid FilterCategory)
at osu.Framework.Input.CameraManager.EnumerateAllDevices()
at osu.Framework.Input.CameraManager.syncCameraDevices()
at osu.Framework.Input.CameraManager.<.ctor>b__19_1()
at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
```

I tried to run the same command with the artifacts of github CI at it gives me the same error.

With `dotnet test` it doesn't give me any errors or warnings.

OS: Linux Manjaro-5.8.6
dotnet: 3.1.107

**Comment by LeNitrous** [comment]
Transfer to Camera Library.

**Comment by sr229** [comment]
The reason for this is because Camera Enumeration is limited to just Windows, which uses COM objects. I already wrote a cross-platform enumerator but it was dropped shortly after this was removed. 

I found a C++ library we can use but I'm not sure how C# interop works with such.

**Comment by LeNitrous** [comment]
This should be the line responsible for checking devices. We should move away from `DirectShow` as we'll need to support other platforms.
https://github.com/vignette-project/osu-framework-camera/blob/67a6f089a1c410dd5e1f91573d0d65fa38c769dd/osu.Framework.Camera/Input/CameraManager.cs#L51

**Comment by sr229** [comment]
@Fede-26  you will need to build `osu-framework-camera` and see if the camera enumeration works now.

**Comment by Fede-26** [comment]
Now it works and it views my webcam.

I don't know if I need to open a new issue but when I tried to compile the `holotrack.Desktop` and `holotrack.Test` but it gave me the error that `libOpenCvSharpExtern.dll` wasn't found so I added the dll manually in `vignette/holotrack.Tests/bin/Debug/netcoreapp3.1/` and `vignette.old/holotrack.Desktop/bin/Debug/netcoreapp3.1/`.

When I run the test project it works and I can view my webcam. But when I run the desktop project it give me this error: https://paste.ubuntu.com/p/7wjNSpCgtD/
